File: Rust/0001-two-sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut seen: HashMap<i32, i32> = HashMap::new();

        for (index, num) in nums.iter().enumerate() {
            if let Some(other_index) = seen.get(num) {
                return vec![*other_index, index as i32];
            } else {
                seen.insert(target - num, index as i32);
            }
        }

        return vec![-1, -1];
    }
}

File: dev-c++/0001-two-sum.cpp
#include <unordered_map>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> visited;

        for (auto const& [index, num]: nums | std::views::enumerate) {
            if(visited.contains(target - num)) {
                return std::vector { visited[target - num], (int) index };
            }
            visited[num] = index;
        }
 
        return std::vector { -1, -1 };
    }
};